pkg/registry/discovery/endpointslice/strategy.go

```diff
@@ -92,7 +92,11 @@ func (endpointSliceStrategy) Validate(ctx context.Context, obj runtime.Object) f
 
 // WarningsOnCreate returns warnings for the creation of the given object.
 func (endpointSliceStrategy) WarningsOnCreate(ctx context.Context, obj runtime.Object) []string {
-	return nil
+	eps := obj.(*discovery.EndpointSlice)
+	if eps == nil {
+		return nil
+	}
+	return validation.GetWarningsForEndpointSliceCreation(eps)
```

Comment:
Why do we want a new pkg for this?  It seems easier to just move the body of `GetWarningsForEndpointSliceCreation()` here and move `warnOnDeprecatedAddressType` into this file.

Is there a reason to do it in a diff pkg?

Reply:
I though that new file would be better in case new warnings will be created to the EndpointSlices but I agree this is a tiny function so I can move that.

Reply:
And also strategy_test file is getting longer.